test/fixtures/iterator.js

```diff
@@ -0,0 +1,475 @@
+'use strict';
+
+const path = require('path');
+const { fork } = require('child_process');
+
+const metatests = require('metatests');
+const { AsyncIterator, asyncIter } = require('../../');
+
+const array = [1, 2, 3, 4];
+
+metatests.test('new AsyncIterator() on non Iterable', test => {
+  test.throws(() => {
+    new AsyncIterator(2);
+  }, new TypeError('Base is not Iterable'));
+  test.end();
+});
+
+metatests.test('new AsyncIterator() on AsyncIterable', test => {
+  const iterator = array[Symbol.iterator]();
+  const iterable = {
+    [Symbol.asyncIterator]: () => (
+      { next: async() => iterator.next() }
+    ),
+  };
+
+  const iter = asyncIter(iterable);
+  test.assert(iter instanceof AsyncIterator);
+  test.end();
+});
+
+metatests.test('next returns Promise', test => {
+  const iter = asyncIter(array);
+  const item = iter.next();
+
+  item.then(() => test.end());
```

Comment:
It only tests that item is `Thanable` not a `Promise`, moreover it does that implicitly.